index.js

```diff
@@ -1,44 +1,60 @@
-var PNG = require('png-js');
-var charm = require('charm');
-var x256 = require('x256');
-var buffers = require('buffers');
-var es = require('event-stream');
-
-var Stream = require('stream').Stream;
-
-module.exports = function (opts) {
-    if (!opts) opts = {};
-    if (!opts.cols) opts.cols = 80;
-    
-    var c = charm();
-    var bufs = buffers();
-    
-    var ws = es.writeArray(function (err, bufs) {
-        var data = buffers(bufs).slice();
-        var png = new PNG(data);
-        
-        png.decode(function (pixels) {
-            var dx = png.width / opts.cols;
-            var dy = 2 * dx;
-            
-            for (var y = 0; y < png.height; y += dy) {
-                for (var x = 0; x < png.width; x += dx) {
-                    var i = (Math.floor(y) * png.width + Math.floor(x)) * 4;
-                    
-                    var ix = x256([ pixels[i], pixels[i+1], pixels[i+2] ]);
-                    if (pixels[i+3] > 0) {
-                        c.background(ix).write(' ');
-                    }
-                    else {
-                        c.display('reset').write(' ');
-                    }
-                }
-                c.display('reset').write('\r\n');
-            }
-            
-            c.display('reset').end();
-        });
+var PNG = require("png-js");
+var charm = require("charm");
+var x256 = require("x256");
+
+var { Duplex } = require("stream");
+
+class PictureTube extends Duplex {
+  constructor(opts = {}) {
+    super();
+    this.data = [];
+    this.cols = opts.cols || 80;
+
+    this.c = charm();
+
+    this.c.on("data", chunk => {
+      this.push(chunk);
     });
-    
-    return es.duplex(ws, c);
+
+    this.c.on("end", () => {
+      this.emit("end");
```

Comment:
this should be this.push(null) to correctly end the stream.

Reply:
Thanks for jumping in Matteo!

I saw that in the docs as well for a readable stream implementation and had tried `this.push(null);` when `c` streams ends but that just doesn't draw anything at all.

To add context:
Both variations of this.push(null) and this.emit('end') work well with this picture-tuber branch if you test it out with the CLI util: `node bin/tube.js /Users/lirantal/projects/forks/blessed-contrib/examples/media/flower.png`:

![image](https://user-images.githubusercontent.com/316371/56090172-1ec09e00-5ea7-11e9-98f4-475ca82a5dbd.png)

The issue is when I use this branch of picture-tuber in blessed-contrib (its used in the [picture widget](https://github.com/yaronn/blessed-contrib/blob/master/lib/widget/picture.js)). To re-construct that setup:
* `cd picture-tuber`
* `git checkout feat/refactor-streams-use`
* `npm link`
* `cd blessed-contrib`
* `npm link picture-tuber`
* `cd examples`
* `node picture.js`